dubbo-common/src/main/java/org/apache/dubbo/rpc/model/ApplicationModel.java

```diff
@@ -234,12 +236,18 @@ public ApplicationConfig getCurrentConfig() {
     }
 
     public String getApplicationName() {
-        return getCurrentConfig().getName();
+        if (StringUtils.isBlank(this.name)) {
+            this.name = getCurrentConfig().getName();
+        }
+        return this.name;
```

Comment:
ApplicationName and ApplicationModel's name are different

Reply:
ok